File: 1904.py
n = int(input())

m = [0 for _ in range(n+1)]
m[0] = 1
m[1] = 1

for i in range(2, n+1):
    m[i] = (m[i-2] + m[i-1]) % 15746

print(m[n])

File: 20250216/1904.cpp
// #include <cstdio>
#include <iostream>
using namespace std;

int n;
const int MOD = 15746;
int bin[1000001];

// 0.75초
int main() {
  ios_base::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);

  // freopen("input.txt", "r", stdin);
  cin >> n;
  bin[1] = 1;
  bin[2] = 2;
  for (int i = 3; i <= n; i++) {
    bin[i] = (bin[i - 2] + bin[i - 1]) % MOD;
  }

  cout << bin[n] << "\n";
  return 0;
}